api/src/main/java/org/openmrs/module/xdssender/api/xds/MessageUtil.java

```diff
@@ -55,18 +58,19 @@ public ProvideAndRegisterDocumentSetRequestType createProvideAndRegisterDocument
 		registryRequest.setRegistryObjectList(new RegistryObjectListType());
 		ExtrinsicObjectType oddRegistryObject = new ExtrinsicObjectType();
 		// ODD
-		oddRegistryObject.setId("Document01");
+		oddRegistryObject.setId(encounter.getLocation().toString() + ":" + patientIsantePlusId + ":"
```

Comment:
I prefer if you use the patientEcidId here instead of the patientIsantePlusId.

Also, can we check if the getLocation() returns the name of the location? I think that's what we would want for the metadata.

Reply:
I checked it before and getLocation() method returns name of the location.